### PR TITLE
Add more test cases

### PR DESCRIPTION
**Pull request by haoel**
Improvements
- Add test cases for the `channel`, `conf`, `report`, `probe` package
- Add test cases for the `ssh` and `http` probe
- Using go 1.19 for lint
- Using atomic for set/get `DryNotify` variable

### CODE REVIEW

File: channel/manager.go [hunk 1]
```diff
@@ -19,18 +19,28 @@ package channel
...
+	"sync/atomic"
 
 	"github.com/megaease/easeprobe/notify"
 	"github.com/megaease/easeprobe/probe"
 )
 
 var channel = make(map[string]*Channel)
 var wg sync.WaitGroup
-var dryNotify bool
+var dryNotify atomic.Value
```
[the remark this answers is not in the document]
Comment (haoel) [reply channel/manager.go:30]: At this moment, I think it's better to be compatible with go 1.18. As the go 1.19 is new, could have some bugs.

File: channel/channel.go [hunk 2]
```diff
@@ -159,7 +160,7 @@ func (c *Channel) WatchEvent(wg *sync.WaitGroup) {
 			log.Infof("[%s / %s]: %s (%s) - Status changed [%s] ==> [%s]",
 				kind, c.Name, result.Name, result.Endpoint, result.PreStatus, result.Status)
 			for _, n := range c.Notifiers {
-				if dryNotify {
+				if GetDryNotify() == true {
```
[the remark this answers is not in the document]
Comment (haoel) [reply channel/channel.go:163]: I'd like to use this kind of style. it's  explicit to read. 

`if flag == true   { ... } `   is more readable then `if flag { ...}`
`if flag == false { ... }`  is better then `if !flag { ... }`

File: report/types_test.go [hunk 3]
```diff
@@ -65,3 +65,10 @@ func TestFormatYAML(t *testing.T) {
 	testFormatYAML(t, SMS, "sms\n")
 	testFormatYAML(t, Unknown, "unknown\n")
 }
+
+func TestBadFormat(t *testing.T) {
+	buf := ([]byte)("- asdf::")
+	var f Format
+	err := yaml.Unmarshal(buf, &f)
+	assert.NotNil(t, err)
+}
```
Comment (zhao-kun) [inline report/types_test.go:74]: Why do we need this? it hasn't relevant to our codes

Comment (haoel) [reply report/types_test.go:74]: I don't understand your comment.  The `Format` is a type that could have marshal/unmarshal operations. it's relevant. and this test case just tests the bad yaml format that causes the unmarshal error.

Comment (zhao-kun) [reply report/types_test.go:74]: got it

File: channel/manager.go [hunk 4]
```diff
@@ -19,18 +19,28 @@ package channel
...
+}
 
 // SetDryNotify sets the global dry run flag
 func SetDryNotify(dry bool) {
-	dryNotify = dry
+	dryNotify.Store(dry)
+}
+
+// GetDryNotify returns the dry run flag
+func GetDryNotify() bool {
```
Comment (zhao-kun) [inline channel/manager.go:42]: I think `IsDryNotify` is a better naming.

Comment (haoel) [reply channel/manager.go]: done!

**Review by localvar — APPROVED** [review]

**Review by zhao-kun — APPROVED** [review]

**Pull request merged**